src/main/java/org/broadinstitute/hellbender/tools/walkers/coverage/DepthOfCoverage.java

```diff
@@ -0,0 +1,445 @@
+            if (!omitIntervals) {
+                for (DoCOutputType.Partition p : partitionTypes) {
+                    // Write the per-interval depth information as necessary
+                    final DepthOfCoverageStats coverageByAggregationPartitionType = partitionerToRemove.getCoverageByAggregationType(p);
+
+                    // Create a new table if necessary
+                    if (!perIntervalStatisticsAggrigationByPartitioning.containsKey(p)) {
+                        perIntervalStatisticsAggrigationByPartitioning.put(p, new int[coverageByAggregationPartitionType.getHistograms().size()][coverageByAggregationPartitionType.getEndpoints().length + 1]);
+                    }
+                    // Update the target table to reflect the updated coverage information for this target
+                    CoverageUtils.updateTargetTable(perIntervalStatisticsAggrigationByPartitioning.get(p), coverageByAggregationPartitionType);
+                }
+            }
+
+        } else if (activeInterval instanceof RefSeqFeature) {
+            DepthOfCoverageStats coverageBySample = partitionerToRemove.getCoverageByAggregationType(DoCOutputType.Partition.sample);
+
+            if ( ! omitPartiallyCoveredGenes || ((RefSeqFeature)activeInterval).getTotalExonLength() <= coverageBySample.getNumberOfLociCovered()) {
+                writer.writePerGeneDepthInformation((RefSeqFeature) activeInterval, coverageBySample, globalIdentifierMap.get(DoCOutputType.Partition.sample));
+            }
+        } else {
+            throw new GATKException("Unrecognized Locatable object supplied for traversal, only RefSeqFeature and SimpleInterval are supported: "+activeInterval.toString());
+        }
+    }
+
+    /**
+     * When we finish traversal we should have the following held in state:
+     *  1. {@link #perIntervalStatisticsAggrigationByPartitioning} that contains a map from partition to a doubly indexed integer array
+     *     that corresponds to a count of the number of data sources with at least the given depth. Importantly this is
+     *     expected to have had CoverageUtils.updateTargetTable() called on it once per interval (i.e. once per locus).
+     *
+     *  2. {@link #coverageTotalsForEntireTraversal} Which we expect to have had update() called on it exactly once for
+     *     every locus traversed by the locus walker. Consequently we expect partitioner to contain partitioned counts
+     *     corresponding to the entire window covered by the tool.
+     *
+     */
+    @Override
+    public Object onTraversalSuccess() {
+        // Write out accumulated interval summary statistics
+        for (DoCOutputType.Partition partition : perIntervalStatisticsAggrigationByPartitioning.keySet()) {
+            writer.writeOutputIntervalStatistics(partition, perIntervalStatisticsAggrigationByPartitioning.get(partition), CoverageUtils.calculateCoverageHistogramBinEndpoints(start,stop,nBins));
```

Comment:
It would seem that way, but i will point out that each partition owns its own output stream independent of all of the others. This means that we might write them out in an undefined order but in reality it doesn't matter because they are all getting written to a different place.